piptools/scripts/compile.py

```diff
@@ -301,6 +301,13 @@ def _determine_linesep(
     help="Specify a package to consider unsafe; may be used more than once. "
     f"Replaces default unsafe packages: {', '.join(sorted(UNSAFE_PACKAGES))}",
 )
+@click.option(
+    "--override-environment",
+    multiple=True,
+    type=(str, str),
```

Comment:
can we add validation on the values that people can override? I would expect this to fail if I pass `--override-environment foo bar` to the CLI. If this can be shown in the `help` to tell the user what environment markers can be overridden, it would improve the experience as well.

Reply:
Added argument validation